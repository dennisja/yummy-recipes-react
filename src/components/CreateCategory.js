import React from 'react';
import { withRouter } from 'react-router-dom';
import AddCategoryForm from './forms/CategoryForm';
import FormCard from './forms/FormCard';

const CreateCategory = () => (
  <div className="row">
    <div className="col m4">
        side bar component here
    </div>
    <div className="col m8">
      <FormCard form={<AddCategoryForm />} title="Add New Category" />
    </div>
  </div>
);

// const CreateCategoryWithRouter = withRouter(CreateCategory);
export default CreateCategory;
