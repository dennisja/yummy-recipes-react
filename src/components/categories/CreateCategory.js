import React from 'react';
import { withRouter } from 'react-router-dom';
import AddCategoryForm from '../forms/CategoryForm';
import FormCard from '../forms/FormCard';
import SideBar from '../SideBar';

const CreateCategory = props => (
  <div className="row">
    <div className="col m4">
      <SideBar />
    </div>
    <div className="col m8">
      <FormCard form={<AddCategoryForm {...props} />} title="Add New Category" />
    </div>
  </div>
);

export default CreateCategory;
