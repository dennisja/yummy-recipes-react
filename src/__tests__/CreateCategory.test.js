import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import CreateCategory from '../components/categories/CreateCategory';

describe('Test CreateCategory component', () => {
    const categoryWrapper = shallow(<CreateCategory />);
    it('renders properly', () => {
        expect(shallowToJson(categoryWrapper)).toMatchSnapshot();
    });

    it('renders a side bar component', () => {
        expect(categoryWrapper.find('SideBar').length).toBe(1);
    });

    it('renders a form card', () => {
        expect(categoryWrapper.find('FormCard').length).toBe(1);
    });
});
