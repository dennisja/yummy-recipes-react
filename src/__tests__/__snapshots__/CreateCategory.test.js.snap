// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CreateCategory component renders properly 1`] = `
<div
  className="row"
>
  <div
    className="col m4"
  >
    <SideBar />
  </div>
  <div
    className="col m8"
  >
    <FormCard
      form={
        <AddCategoryForm
          role="add"
        />
      }
      iconClass={null}
      title="Add New Category"
    />
  </div>
</div>
`;
